### PR TITLE
Styling for sales sidebar on checkout screen + alphabet scrollbar

### PR DESCRIPTION
**Pull request by thumn**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Added an alphabetical scroll bar to the bottom of the checkout screen that allows you to scroll to the first product with a name in that letter range.
- Restyled checkout screen's sales container with (1) checkout button, (2) components with subtotal price, rewards applied, total price, and points earned, and (3) spacing between components.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
N/A

### Online sources
[Pretty helpful article on how to create buttons that scroll to different parts on a ScrollView](https://medium.com/@ayushi.nig/autoscroll-in-react-native-451601ac3ca8)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
#11 

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
Use checkout bypass to see if checkout flow works for you!
- Add several items to the cart to see if spacing on the sidebar between the cart and rewards/total price components works for you
- See if the alphabet scrollbar height matches (roughly) the checkout button height
- Should be about five columns of products on a tablet view (this is how I calculate where to scroll...)
- Test out the alphabet scroll tabs -- make sure they go to the right first items in the range

Note: Optimized for viewing on a tablet in landscape mode. Portrait mode/viewing on a phone is kinda jank rn...

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
Post-MVP plan is to make this compatible for mobile view. This would involve converting all hard-coded widths for product cards and line item cards, reworking the number of columns to display and/or playing around with text base components, and testing for visibility (since there is a lot of content on one screen as of right now).

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
Mostly end-to-end testing in terms of logging in as a clerk, looking up a customer, creating a large transaction, utilizing scrollbar tabs, and checking out. It works if the transaction shows up on Airtable!

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
IMG_70F3E953A4CE-1

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: lib/checkoutUtils.js [hunk 1]
```diff
@@ -82,3 +82,9 @@ export function displayDollarValue(amount, positive = true) {
   }
   return '-'.concat(dollarValue);
 }
+
+// TODO: change back to 5
```
Comment (anniero98) [inline lib/checkoutUtils.js]: Is this being used? :o

File: screens/CheckoutScreen.js [hunk 2]
```diff
@@ -181,61 +184,113 @@ export default class CheckoutScreen extends React.Component {
...
+  };
+
+  // Takes in strings tab label (i.e. "A-K") and starting letter (i.e. "A") and returns a
+  // tab for the bottom alphabetical scroll bar.
+  alphabeticalScrollTab = (label, letter) => {
+    return (
+      <TabContainer
+        onPress={() =>
+          this._scrollView.scrollTo({
+            y: Math.floor(this.getIndexOfFirstProductAtLetter(letter) / 5) * 160
```
Comment (anniero98) [inline screens/CheckoutScreen.js]: ah, I can see why making this responsive would be very rough 😂

File: screens/CheckoutScreen.js [hunk 3]
```diff
@@ -181,61 +184,113 @@ export default class CheckoutScreen extends React.Component {
...
+  // Returns index of the first product with a name starting with the given letter in products list.
+  // If no product starting with that letter exists, find the next product.
+  getIndexOfFirstProductAtLetter = letter => {
+    let prodList = this.state.products.filter(product => product.name.charAt(0) === letter);
+    let nextLetter = letter;
+    while (prodList.length === 0) {
+      nextLetter = String.fromCharCode(nextLetter.charCodeAt(0) + 1);
+      prodList = this.state.products.filter(product => product.name.charAt(0) === nextLetter);
+    }
+    return this.state.products.indexOf(prodList[0]);
```
Comment (anniero98) [inline screens/CheckoutScreen.js]: Nice!! A couple of things, which I think could make this a little more compact (and handle some edge cases):
- Javascript supports comparison with ASCII values, so for alphabetical ordering you can compare character values with `<` and `>`
- Should account for both first letter and last letter, since it's possible that once we start filtering the product lists to be per-store only (very few products in a store), there will be empty tabs?
- Should use `.toLowercase` or `.toUppercase` depending on what you're passing in for 'firstLetter' and 'lastLetter' when comparing, just for robustness (in case an admin mistakenly changes the name of a product)
```
// all products that start with letters between these
let prodList = this.state.products.filter(product => product.name.charAt(0).toLowerCase() <= startLetter && product.name.charAt(0) > endLetter);
if (prodList.length === 0) {
 // error check, since when we start filtering products there might be none that exist
// not rly sure rn but have to return something else
}
return this.state.products.indexOf(prodList[0]);
```

^ not sure if this actually works but you get the idea!

File: screens/CheckoutScreen.js (hunk 2 above)
Comment (anniero98) [reply screens/CheckoutScreen.js]: I'm curious if this math will still work once the product list shrinks due to being store-specific?

File: styled/checkout.js [hunk 4]
```diff
@@ -15,6 +15,30 @@ export const TopBar = styled.View`
...
 `;
 
+// Alphabetical Tab Bottom Bar
+// TODO: flex value fixes height of alphabetical bar to match height of checkout button. Is there a better way to do this?
+
+export const BottomBar = styled.View`
+  height: 57px;
+  width: 100%;
+  background: ${Colors.base};
+  flex: 0.44;
```
Comment (wangannie) [inline styled/checkout.js]: I'm curious what does this do? When I take it out nothing seems to be affected.

File: screens/CheckoutScreen.js [hunk 5]
```diff
@@ -181,61 +184,113 @@ export default class CheckoutScreen extends React.Component {
...
-              Order Total ${totalSale.toFixed(2)}
-            </Text>
-            <TouchableOpacity onPress={() => this.handleSubmit()}>
-              <TextHeader style={{ color: '#008550' }}>Checkout</TextHeader>
-            </TouchableOpacity>
+              <View style={{ height: '60%' }}>
+                <Subhead>Current Sale</Subhead>
+                {/* Cart container */}
+                <View style={{ height: '100%', paddingBottom: '5%' }}>
+                  <ScrollView>
```
Comment (wangannie) [inline screens/CheckoutScreen.js]: UX detail: Can you add something where it scrolls to the bottom of the ScrollView on size change (aka whenever a new product is added to this list)? Otherwise, it lacks that kind of confirmation that the product was added and could cause some confusion.
Something like this: https://stackoverflow.com/a/42736127

File: screens/CheckoutScreen.js [hunk 6]
```diff
@@ -181,61 +184,113 @@ export default class CheckoutScreen extends React.Component {
...
+                  callback={this.applyRewardsCallback}
+                />
+                {/* When different types of rewards are created, we can add rewards amount to state. For now, rewards
+              amount is equal to rewards applied * 5. */}
+                <SubtotalCard subtotalPrice={subtotal.toFixed(2)} rewardsAmount={this.state.rewardsApplied * 5} />
+                <TotalCard totalPrice={totalSale.toFixed(2)} totalPoints={pointsEarned} />
+              </View>
+            </View>
+            <FilledButtonContainer onPress={() => this.handleSubmit()}>
+              <ButtonLabel>Complete Purchase</ButtonLabel>
```
Comment (wangannie) [inline screens/CheckoutScreen.js]: Design update: Ace wanted to keep the language consistent across the app (since it is currently 'Current Sale' in the heading, 'Complete Purchase' on the button, 'Confirm Transaction' on the system dialog, then 'Purchase Summary' on the confirmation), so can we change everything to Sale? 

Complete Purchase -> Complete Sale
Purchase Summary -> Sale Summary
Confirm Transaction -> Confirm Sale

Comment (wangannie) [reply screens/CheckoutScreen.js]: This is a recent update but should be reflected in Figma

File: components/TotalCard.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @prop
+ **/
+
+// TODO @tommypoa to use styled-components
+function TotalCard({ totalPrice, totalPoints }) {
+  return (
+    <LineItem>
+      <LineItemRow>
+        <Subhead>Total</Subhead>
```
Comment (wangannie) [inline components/TotalCard.js]: Don't forget to change this to Total Sale

File: screens/CheckoutScreen.js [hunk 8]
```diff
@@ -181,61 +188,113 @@ export default class CheckoutScreen extends React.Component {
...
+                this._scrollView = scrollView;
+              }}>
+              {Object.entries(cart).map(([id, product]) => (
+                <QuantityModal key={id} product={product} isLineItem={false} callback={this.updateQuantityCallback} />
+              ))}
+            </ProductsContainer>
+            <BottomBar style={{ display: 'flex', flexDirection: 'row' }}>
+              {this.alphabeticalScrollTab('A-K', 'A')}
+              {this.alphabeticalScrollTab('L-S', 'L')}
+              {this.alphabeticalScrollTab('T-Z', 'T')}
```
Comment (wangannie) [inline screens/CheckoutScreen.js]: Can we add spaces between the hyphens like 'A - K' so it looks less cramped? (Ace approved)

File: screens/CheckoutScreen.js (hunk 2 above)
Comment (thumn) [reply screens/CheckoutScreen.js]: @anniero98 I talked to @wangannie about this and I think we ended up deciding that clerks would always see all products on the checkout screen? (I think it was because maybe they don't update the products available at a certain store in the Airtable before giving it to them physically but correct me if I'm wrong)

Comment (wangannie) [reply screens/CheckoutScreen.js]: Yeah I don't think it's going to be store specific

File: styled/checkout.js (hunk 4 above)
Comment (thumn) [reply styled/checkout.js]: omg wat it works without -- amazing
thank u for catching this, it was making me nervous :' )
